distributed/protocol/h5py.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import print_function, division, absolute_import
+
+import h5py
+
+from .serialize import register_serialization
+
+from ..utils import log_errors, ensure_bytes
+
+
+def serialize_h5py_file(f):
+    return {'filename': f.filename, 'mode': f.mode}, []
```

Comment:
Why not just restrict it to reading outright? It seems like `r+` has the same risks attached as a writable files.